File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppJava/app/src/main/java/pl/udu/uwr/pum/verynobleappjava/adapters/nobelprizelaureates/LaureateComparator.java
package pl.udu.uwr.pum.verynobleappjava.adapters.nobelprizelaureates;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import pl.udu.uwr.pum.verynobleappjava.data.nobelprizeresponse.NobelAwardsResponse;

public class LaureateComparator extends DiffUtil.ItemCallback<NobelAwardsResponse.NobelPrizes.Laureates> {
    @Override
    public boolean areItemsTheSame(@NonNull NobelAwardsResponse.NobelPrizes.Laureates oldItem, @NonNull NobelAwardsResponse.NobelPrizes.Laureates newItem) {
        return oldItem.getId().equals(newItem.getId());
    }

    @Override
    public boolean areContentsTheSame(@NonNull NobelAwardsResponse.NobelPrizes.Laureates oldItem, @NonNull NobelAwardsResponse.NobelPrizes.Laureates newItem) {
        return newItem.getId().equals(oldItem.getId());
    }
}

File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppKotlin/app/src/main/java/pl/udu/uwr/pum/verynobleappkotlin/adapters/nobleprizelaureate/LaureateComparator.kt
package pl.udu.uwr.pum.verynobleappkotlin.adapters.nobleprizelaureate

import androidx.recyclerview.widget.DiffUtil
import pl.udu.uwr.pum.verynobleappkotlin.data.nobelprizeresponse.laureate.Laureate

class LaureateComparator : DiffUtil.ItemCallback<Laureate>() {
    override fun areItemsTheSame(oldItem: Laureate, newItem: Laureate): Boolean {
        return oldItem.id == newItem.id
    }

    override fun areContentsTheSame(oldItem: Laureate, newItem: Laureate): Boolean {
        return oldItem == newItem
    }
}